common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidPlatformUtil.java

```diff
@@ -131,7 +134,34 @@ public void removeCookiesFromWebView() {
 
     @Override
     public boolean isValidCallingApp(@NonNull String redirectUri, @NonNull String packageName) {
-        return BrokerValidator.isValidBrokerRedirect(redirectUri, mContext, packageName);
+        final String methodTag = TAG + ":isValidCallingApp";
+        final String expectedBrokerRedirectUri = PackageHelper.getBrokerRedirectUri(mContext, packageName);
+        boolean isValidBrokerRedirect = StringUtil.equalsIgnoreCase(redirectUri, expectedBrokerRedirectUri);
+        if (packageName.equals(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)) {
+            final PackageHelper info = new PackageHelper(mContext.getPackageManager());
+            //For merely verifying that the app is AuthApp, use a 512 hash.
+            final String signatureDigest = info.getSha512SignatureForPackage(packageName);
```

Comment:
I should probably rename the getSha1/512SignatureForPackage methods to getSha512HashForPackage, or getSha512ThumbprintForPackage, to match with the naming changes in this PR.

Reply:
I'll just do additional renaming in these PRs.

Reply:
decide to keep this one for now, as i'm avoiding making breaking change to ADAL.